components/block/element.js

```diff
@@ -9,56 +9,93 @@ export default class Block extends LitElement {
   static styles = [style];
 
   static properties = {
-    blockId: { type: Number},
+    blockId: {type: String, attribute: 'data-ngl-block-id'},
   };
 
-  renderMenu(){
-    return html`
-      <nav>
-        <button @click=${this.edit}>Edit</button>
-        <button @click=${this.moveUp}>Move Up</button>
-        <button @click=${this.moveDown}>Move Down</button>
-        <button @click=${this.delete}>Delete</button>
-      </nav>
-    `;
-  }
+  constructor() {
+    super();
 
-  edit(e){
-    console.log(e.target.innerText)
-    this.refresh();
+    this.isSaveListenerAttached = false;
   }
 
-  delete(e) {
-    console.log(e.target.innerText)
-    this.remove();
+  getBlock() {
+    return window.parent.Core.g.layout.blocks.findWhere({id: this.blockId});
   }
 
-  moveUp(e) {
-    console.log(e.target.innerText)
-    const prevElement = this.previousElementSibling;
-    isBlock(prevElement) && this.parentNode.insertBefore(this, prevElement)
+  renderMenu() {
+    const block = this.getBlock();
+    const isInLinkedZone = block.zone().is_linked();
+
+    if (isInLinkedZone) {
+      return html`
+        <div class="edit-menu">
+          <button @click=${this.refresh}>Refresh</button>
+        </div>
+      `;
+    } else {
+      const parentId = block.attributes.parent_block_id;
+      const parent = document.querySelector(
+        `ngl-block[data-ngl-block-id="${parentId}"]`
+      );
```

Comment:
extract to getter

Reply:
Fixed in 506394a14afbb78b493e59d4dc5f6fc4a04890d8